bin/reth/src/args/types.rs

```diff
@@ -36,6 +36,28 @@ impl FromStr for ZeroAsNone {
     }
 }
 
+/// A helper type for parsing "max" as `u32::MAX`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct MaxU32(pub u32);
```

Comment:
cool, now since there is no trait in rust std that allows us to get `MAX`

we want this for u64 as well, I recommend a simple macro for this that generates the code

```rust
macro_rules! max_values($name:ident, $ty:ident) {
   pub struct Max$name(pub $ty);
   ....
}

max_values!(U32, u32);

```

Reply:
> cool, now since there is no trait in rust std that allows us to get `MAX`
> 
> we want this for u64 as well, I recommend a simple macro for this that generates the code
> 
> ```rust
> macro_rules! max_values($name:ident, $ty:ident) {
>    pub struct Max$name(pub $ty);
>    ....
> }
> 
> max_values!(U32, u32);
> ```

ty, does this looks better?